demo-projects/meetup/site/pages/events.js

```diff
@@ -2,39 +2,33 @@
 
 import { jsx } from '@emotion/core';
 
-import { useQuery } from '@apollo/react-hooks';
-
-import { Container, Loading, H2 } from '../primitives';
+import { Container, H2 } from '../primitives';
 import EventItems from '../components/EventItems';
 import Navbar from '../components/Navbar';
 import Footer from '../components/Footer';
 import Meta from '../components/Meta';
 import { gridSize } from '../theme';
 
 import { GET_ALL_EVENTS } from '../graphql/events';
+import { initApolloClient } from '../lib/apollo';
 
-export default function Events() {
-  const { data: { allEvents } = {}, loading, error } = useQuery(GET_ALL_EVENTS);
-
-  if (error) {
-    console.error('Failed to load events', error);
-  }
-
+export default function Events({ allEvents }) {
   return (
     <>
       <Meta title="Events" />
       <Navbar background="white" />
       <Container css={{ marginTop: gridSize * 3 }}>
         <H2>Events</H2>
-        {loading ? (
-          <Loading isCentered size="xlarge" />
-        ) : error ? (
-          <p>Something went wrong. Please try again.</p>
-        ) : (
-          <EventItems events={allEvents} />
-        )}
+        <EventItems events={allEvents} />
       </Container>
       <Footer />
     </>
   );
 }
+
+export async function getStaticProps() {
+  const apolloClient = initApolloClient();
+  const { data } = await apolloClient.query({ query: GET_ALL_EVENTS });
+
+  return { props: { allEvents: data.allEvents } };
```

Comment:
This differs for the examples I've seen in the the NextJS repo that use `apolloClient.cache.extract()` to set `initialApolloState`: https://github.com/vercel/next.js/blob/canary/examples/with-apollo/pages/index.js#L30

I believe this initial state is then used by the provider to warm the cache: https://github.com/vercel/next.js/blob/canary/examples/with-apollo/lib/apolloClient.js#L24

You then use queries on the client as normal and the data will be available without the need to a network request.

My experience with this is not extensive yet so I could be missing something but happy to share some examples that appear to be working.